cmd/systray/systray.go

```diff
@@ -425,25 +443,14 @@ func (menu *Menu) rebuildExitNodeMenu(ctx context.Context) {
 	time.Sleep(newMenuDelay)
 
 	// register a click handler for a menu item to set nodeID as the exit node.
-	onClick := func(item *systray.MenuItem, nodeID tailcfg.StableNodeID) {
-		go func() {
-			for {
-				select {
-				case <-ctx.Done():
-					return
-				case <-item.ClickedCh:
-					select {
-					case <-ctx.Done():
-						return
-					case menu.exitNodeCh <- nodeID:
-					}
-				}
-			}
-		}()
+	setExitNodeOnClick := func(item *systray.MenuItem, nodeID tailcfg.StableNodeID) {
+		onClick(ctx, item, func() {
+			menu.exitNodeCh <- nodeID
```

Comment:
if you had ctx, you could select on this send + ctx.Done recv?

Reply:
done